dss_datamover/monitor.py

```diff
@@ -277,13 +279,33 @@ def monitor_index_distributor(self):
                         self.received_index_msg_count.value += 1
 
                     object_count += object_count_under_prefix
+            
+            # resend failed data
+            elif len(resend_index_data) > 0:
+                data, last_client_index, start_time = resend_index_data.pop(0)
+                if (datetime.now() - start_time).seconds > RESEND_WAIT_TIME:
```

Comment:
since the failed data will stop being resent once we go past RESEND_WAIT_TIME do we need to mark another flag to shut down the monitor?

Reply:
Right now the monitor can stop itself when resend_index_data is empty, but it does no harm to add another flag to notify the master application what's wrong.

Reply:
ok that is not needed, maybe just add some logging to indicate what has occured